pravega-cli/src/main/java/io/pravega/tools/pravegacli/commands/controller/ControllerDescribeStreamCommand.java

```diff
@@ -122,18 +131,42 @@ public static CommandDescriptor descriptor() {
     }
 
     private SegmentHelper instantiateSegmentHelper(CuratorFramework zkClient) {
+
+        // hostMonitorEnable should be true for pks clusters
         HostMonitorConfig hostMonitorConfig = HostMonitorConfigImpl.builder()
-                                                                   .hostMonitorEnabled(true)
+                                                                   .hostMonitorEnabled(false)
+                                                                   .hostContainerMap(getHostContainerMap(getCLIControllerConfig().getSegmentStoreURI(),
+                                                                           getServiceConfig().getContainerCount()))
                                                                    .hostMonitorMinRebalanceInterval(Config.CLUSTER_MIN_REBALANCE_INTERVAL)
                                                                    .containerCount(getServiceConfig().getContainerCount())
                                                                    .build();
         HostControllerStore hostStore = HostStoreFactory.createStore(hostMonitorConfig, StoreClientFactory.createZKStoreClient(zkClient));
         ClientConfig clientConfig = ClientConfig.builder()
                                                 .controllerURI(URI.create((getCLIControllerConfig().getControllerGrpcURI())))
-                                                .validateHostName(getCLIControllerConfig().isAuthEnabled())
-                                                .credentials(new DefaultCredentials(getCLIControllerConfig().getPassword(), getCLIControllerConfig().getUserName()))
+                                                .validateHostName(false)
+                                                .credentials(new DefaultCredentials(getCLIControllerConfig().getPassword(),
+                                                        getCLIControllerConfig().getUserName()))
                                                 .build();
         ConnectionPool pool = new ConnectionPoolImpl(clientConfig, new SocketConnectionFactoryImpl(clientConfig));
         return new SegmentHelper(pool, hostStore, pool.getInternalExecutor());
     }
+
+    private static Map<Host, Set<Integer>> getHostContainerMap(List<String> uri, int containerCount) {
+        Exceptions.checkNotNullOrEmpty(uri, "uri");
+
+        Map<Host, Set<Integer>> hostContainerMap = new HashMap<>();
+        uri.forEach(x -> {
+
+            // Get the host and port from the URI
+            String host = x.split(":")[0];
+            int port = Integer.parseInt(x.split(":")[1]);
+
+            Exceptions.checkNotNullOrEmpty(host, "host");
```

Comment:
Same here, use `Preconditions.checkNotNull`.

Reply:
Changed to `Preconditions.checkNotNullOrEmpty` as suggested.